### ISSUE TITLE
Vascular plant recording form - toggle for use of map not working?

### ISSUE THREAD

**Issue by kitenetter** [BiologicalRecordsCentre/iRecord#1540]
On the vascular plant form, when you move to the species list page, there is an attribute for "Spatial ref" that has an icon next to it. The hover-over text says this icon should allow you to toggle to make a map visible to enter a precise grid ref. However, clicking on the icon does not appear to do anything (tested using Edge).

**Comment by Sam-Amy** [comment]
The inability to add individual grid references to observations on this form was also raised in a [forum post](https://irecord.org.uk/forum/feedback-and-suggestions/data-entry) some time ago.

**Comment by Sam-Amy** [comment]
@burkmarr along with the other changes I've been making to the bryophyte form, we'd also like to implement this functionality - but need to get it working first. Do you think this is something you might have time to look into?

**Comment by burkmarr** [comment]
I've just tried it and it looks as though once toggle on, you need to go back to the grid square tab and use the map that's there.

**Comment by johnvanbreda** [comment]
@burkmarr have you checked the option spatialRefPerRowUseFullscreenMap?

**Comment by Sam-Amy** [comment]
> I've just tried it and it looks as though once toggle on, you need to go back to the grid square tab and use the map that's there.

Is this the intention? I was expecting a map to pip up. Also, when I do this, it doesn't seem to allow me to get a higher resolution than 1km square.

> @burkmarr have you checked the option spatialRefPerRowUseFullscreenMap?

I've just tried adding this to the Form structure code (it wasnt there), if that's what you mean, but this doesn't seem to change anything so I'll remove again.

**Comment by burkmarr** [comment]
@johnvanbreda -  have now! @Sam-Amy @kitenetter - is looks like we just needed to add `@spatialRefPerRowUseFullscreenMap=true`

**Comment by burkmarr** [comment]
@Sam-Amy - I think we were editing the form at the same time

**Comment by Sam-Amy** [comment]
The map is coming up now as expected, but I still can't add a grid reference at a finer resolution that 1km, thich seems the main aim with this function. Is there a way to enable this please @burkmarr @johnvanbreda?

**Comment by burkmarr** [comment]
I suspect that this is because the `@maxGridRef=4` is set. That means that the maximum precision allowed for the the sample is 1 km square, but it must apply this to the sub-samples too.

**Comment by sacrevert** [comment]
I don't actually know when the toggle map icon for the finer scale grid ref was added, I don't remember ever requesting this (but may be forgetting). Originally adding a fine scale grid ref just allowed one to type it in, and the grid ref was displayed on the review and submit page within the 1 or 2 km square chosen for the survey (you can just about see that behaviour on the example here: https://bsbi.org/wp-content/uploads/dlm_uploads/iRecord_BSBI_poster_v2_A1.pdf)

I'm not fussed about the popup map element of the fine-scale grid ref, as long as they are mapped on the review and submit page (and that page zooms in to the 1 or 2 km scale to show the finer-scale refs clearly). My main concerns would be that the finer scale occurrence ref is forced to be in the same grid format as the larger scale square, and that is is validated as a correct grid ref within that format. I don't think selecting via a pop-up map is really that useful, as the field is designed for people taking GPS readings for rare species, and I think this would typically be copied from records elsewhere that were made in the field.

**Comment by Sam-Amy** [comment]
@maxGridRef=4 doesn't prevent entering manually and saving a finer scale grid reference, but with @spatialRefPerRowUseFullscreenMap=true (i.e. pop-up map enabled for observation spatial ref), although you can see there your record shown with a finer scale grid reference, if you then click on that map the grid ref for that record reverts to a 1km square.

This is confusion and could cause errors, so I have removed @spatialRefPerRowUseFullscreenMap=true, so the pop-up doesn't work - as Oli says it is fine to just use the review tab for this. A couple of questions please @burkmarr 

1) Do you know of a way to actually remove the toggle symbol from the observation Spatial Ref field, as it is defunct now?
2) Do you know why the 'GPS precision (m)' field is greyed out and showing n/a once you have entered something in the Spatial ref field (but not before, and if you then add it, the GPS precision changes to n/s)?:
![image](https://github.com/BiologicalRecordsCentre/iRecord/assets/104146164/05ecf714-25af-47d2-8881-34921adfa3f0)

**Comment by burkmarr** [comment]
@Sam-Amy - Looking at the code for the `@spatialRefPerRow` option, it looks like there is no way to turn off the button to pick a GR from a map (the pin symbol). So this would require a code change to implement, e.g. to define a further option such as `@spatialRefPerRowNoMapButton` which would suppress the button.

Looking at the code for the precision problem, there is a check that disables the precision input and enters 'n/a' (presumably meaning 'not applicable') if there are less that 10 digits in the grid ref. In other words you can only enter a value for precision if you supply a ten figure grid ref. That is confusing. The only explanation I can imagine is that it is to discourage people blindly copying down a 10 figure grid ref (which implies a precision of 1 m on the ground) from their GPS without any thought to precision. I suppose that if someone is entering a grid reference of 8 figure or less precision, the assumption is that they are making a judgement about precision which is implicit in their grid ref. However there is nothing to stop someone entering a 10 fig grid ref and then ignoring the precision field which doesn't support that explanation. @johnvanbreda - do you have any thoughts on this? (https://github.com/Indicia-Team/media/blob/master/js/addRowToGrid.js#L802)

**Comment by sacrevert** [comment]
@burkmarr yes, the assumption is that <10 fig implies a grid-based precision, but I agree that entering a 10 fig ref should make the precision field mandatory. 

However, in relation to the precision field and the NBN, I was recently told by Will Millard at the NBN that they would remove any precision field supplied, because it is implied by the grid reference. As you have noted, this is obviously not appropriate for 10 fig regs. I don't know what this implies for indicia/NBN data exchange however...

**Comment by Sam-Amy** [comment]
Thanks @burkmarr, I hadn't noticed that a 10-figure grid reference enabled the precision field. I see the logic now but as you say @sacrevert this should perhaps be mandatory, though I imagine this could also be rather annoying for the user (I don't know the precision of my phones GPS, though perhspa I should!). Also, how often do you think 10-figure grid references are actually used for analytical purposes where a lack of precision matters, and if very rarely then is this a moot point?

**Comment by sacrevert** [comment]
They are more often used for display that for analysis (i.e. displaying a point with a plotted radius). if a user doesn't know the precision, then they can always truncate to 8 figs. I suppose it might be nice to present this as an automatic option highlighted by a mandatory precision warning (i.e. "You haven't filled in the precision field for your 10 figure GR: do you want to truncate this to 8 figures?"; I appreciate that this might be considerable extra work however

**Comment by johnvanbreda** [comment]
@burkmarr the requirement to only show the precision control when a small grid square is specified came from Plant Portal and the BSBI recording forms according to the GitHub comments, so I think fine to change so that the precision becomes mandatory.

Also for the spatialRefPerRow option you could use CSS to hide the map pin in the grid, rather than make yet more custom options.

**Comment by burkmarr** [comment]
I've added a CSS file for the form which hides the link button as suggested by @johnvanbreda.

@johnvanbreda - is there a supported way to make a control shown in the species grid mandatory? If I can see how to do that, I can probably work out how to make it conditional.

**Comment by burkmarr** [comment]
@johnvanbreda - I think I discovered how to do this, but I still have a question about it. I will make a branch from develop and make a pull request and put the question there.

**Comment by sacrevert** [comment]
Seen working
